File: app/src/main/java/com/worldventures/dreamtrips/wallet/domain/converter/SmartCardInfoToSmartCardDetail.java
package com.worldventures.dreamtrips.wallet.domain.converter;

import com.worldventures.dreamtrips.api.smart_card.association_info.model.SmartCardInfo;
import com.worldventures.dreamtrips.modules.mapping.converter.Converter;
import com.worldventures.dreamtrips.wallet.domain.entity.ImmutableSmartCardDetails;
import com.worldventures.dreamtrips.wallet.domain.entity.SmartCardDetails;

import io.techery.mappery.MapperyContext;

class SmartCardInfoToSmartCardDetail implements Converter<SmartCardInfo, SmartCardDetails> {

   @Override
   public Class<SmartCardInfo> sourceClass() {
      return SmartCardInfo.class;
   }

   @Override
   public Class<SmartCardDetails> targetClass() {
      return SmartCardDetails.class;
   }

   @Override
   public SmartCardDetails convert(MapperyContext mapperyContext, SmartCardInfo smartCardInfo) {
      return ImmutableSmartCardDetails.builder()
            .smartCardId(smartCardInfo.scId())
            .serialNumber(smartCardInfo.serialNumber())
            .bleAddress(smartCardInfo.bleAddress())
            .revVersion(smartCardInfo.revVersion())
            .wvOrderId(smartCardInfo.wvOrderId())
            .nxtOrderId(smartCardInfo.nxtOrderId())
            .orderDate(smartCardInfo.orderDate())
            .build();
   }
}

File: wallet/src/main/java/com/worldventures/wallet/domain/converter/SmartCardInfoToSmartCardDetail.kt
package com.worldventures.wallet.domain.converter

import com.worldventures.core.converter.Converter
import com.worldventures.dreamtrips.api.smart_card.user_association.model.SmartCardInfo
import com.worldventures.wallet.domain.entity.SmartCardDetails
import io.techery.mappery.MapperyContext

internal class SmartCardInfoToSmartCardDetail : Converter<SmartCardInfo, SmartCardDetails> {

   override fun sourceClass() = SmartCardInfo::class.java

   override fun targetClass() = SmartCardDetails::class.java

   override fun convert(context: MapperyContext, source: SmartCardInfo): SmartCardDetails {
      return SmartCardDetails(
            smartCardId = source.scId(),
            serialNumber = source.serialNumber(),
            bleAddress = source.bleAddress(),
            revVersion = source.revVersion(),
            wvOrderId = source.wvOrderId(),
            nxtOrderId = source.nxtOrderId(),
            orderDate = source.orderDate())
   }
}
